### PR TITLE
Fixed: Change Request/Limit colors in Memory/CPU resources graphs to …

### PR DESCRIPTION
**Pull request by ganeshrvel**
…red and green

### CODE REVIEW

File: playbooks/robusta_playbooks/pod_enrichments.py [hunk 1]
```diff
@@ -33,13 +34,24 @@ def pod_graph_enricher(pod_event: PodEvent, params: PodResourceGraphEnricherPara
     limit_lines = []
     if params.display_limits:
         resource_limits = pod_limits(pod)
+        request_limits = pod_requests(pod)
```
Comment (Avi-Robusta) [inline playbooks/robusta_playbooks/pod_enrichments.py:37]: rename request_limits to resource_requests

File: playbooks/robusta_playbooks/pod_enrichments.py [hunk 2]
```diff
@@ -33,13 +34,24 @@ def pod_graph_enricher(pod_event: PodEvent, params: PodResourceGraphEnricherPara
...
         if params.resource_type == "CPU" and resource_limits.cpu > 0:
             limit_line = XAxisLine(label="CPU Limit", value=resource_limits.cpu)
-            limit_lines = [limit_line]
+            limit_lines.append(limit_line)
+            if request_limits.cpu > 0:
+                request_memory_limit_in_bytes = request_limits.cpu
+                limit_line = XAxisLine(label="CPU Request", value=request_memory_limit_in_bytes)
```
Comment (Avi-Robusta) [inline playbooks/robusta_playbooks/pod_enrichments.py:43]: replace 
```
request_memory_limit_in_bytes = request_limits.cpu
limit_line = XAxisLine(label="CPU Request", value=request_memory_limit_in_bytes)
```
with
`limit_line = XAxisLine(label="CPU Request", value=request_limits.cpu)`

File: playbooks/robusta_playbooks/pod_enrichments.py [hunk 3]
```diff
@@ -33,13 +34,24 @@ def pod_graph_enricher(pod_event: PodEvent, params: PodResourceGraphEnricherPara
     limit_lines = []
     if params.display_limits:
         resource_limits = pod_limits(pod)
+        request_limits = pod_requests(pod)
         if params.resource_type == "CPU" and resource_limits.cpu > 0:
             limit_line = XAxisLine(label="CPU Limit", value=resource_limits.cpu)
-            limit_lines = [limit_line]
+            limit_lines.append(limit_line)
+            if request_limits.cpu > 0:
```
Comment (Avi-Robusta) [inline playbooks/robusta_playbooks/pod_enrichments.py:41]: you can have a request without a limit so i would unindent this line
and change the if with
`if params.resource_type == "CPU" and resource_requests.cpu > 0:`

File: playbooks/robusta_playbooks/pod_enrichments.py [hunk 4]
```diff
@@ -33,13 +34,24 @@ def pod_graph_enricher(pod_event: PodEvent, params: PodResourceGraphEnricherPara
...
+                request_memory_limit_in_bytes = request_limits.cpu
+                limit_line = XAxisLine(label="CPU Request", value=request_memory_limit_in_bytes)
+                limit_lines.append(limit_line)
+
         elif params.resource_type == "Memory" and resource_limits.memory > 0:
             memory_limit_in_bytes = resource_limits.memory * 1024 * 1024
             limit_line = XAxisLine(label="Memory Limit", value=memory_limit_in_bytes)
-            limit_lines = [limit_line]
+            limit_lines.append(limit_line)
+            if request_limits.memory > 0:
```
Comment (Avi-Robusta) [inline playbooks/robusta_playbooks/pod_enrichments.py:50]: same as above

File: src/robusta/core/playbooks/container_playbook_utils.py [hunk 5]
```diff
@@ -20,7 +20,13 @@ def create_container_graph(params: ResourceGraphEnricherParams, pod: Pod, contai
         if params.resource_type == "Memory" and limits > 0:
             memory_limit_in_bytes = limits * 1024 * 1024
             limit_line = XAxisLine(label="Memory Limit", value=memory_limit_in_bytes)
-            limit_lines = [limit_line]
+            limit_lines.append(limit_line)
```
Comment (Avi-Robusta) [inline src/robusta/core/playbooks/container_playbook_utils.py:23]: this seems to be missing the CPU limits + requests here
I would refactor this to a function that does all of this logic and returns the lines[] of the limits and call it here and in pod_graph_enricher

File: src/robusta/core/playbooks/prometheus_enrichment_utils.py [hunk 6]
```diff
@@ -186,12 +164,59 @@ def create_chart_from_prometheus_query(
...
+    }
+    chart_values_format = values_format if values_format else ChartValuesFormat.Plain
+    chart.value_formatter = value_formatters[chart_values_format]
+
+    if chart_title:
+        chart.title = chart_title
+    else:
+        chart.title = promql_query
+
+    for plot in plot_list:
```
Comment (Avi-Robusta) [inline src/robusta/core/playbooks/prometheus_enrichment_utils.py:217]: what is this for?

**Review by Avi-Robusta — CHANGES_REQUESTED** [review]
looks good, 
Some small bug in the logic, and a refactoring change

File: playbooks/robusta_playbooks/pod_enrichments.py (hunk 2 above)
Comment (ganeshrvel) [reply playbooks/robusta_playbooks/pod_enrichments.py]: I had missed the conversion to bytes in the first place. This block should be fine now.

File: playbooks/robusta_playbooks/pod_enrichments.py (hunk 3 above)
Comment (ganeshrvel) [reply playbooks/robusta_playbooks/pod_enrichments.py]: done

File: playbooks/robusta_playbooks/pod_enrichments.py (hunk 4 above)
Comment (ganeshrvel) [reply playbooks/robusta_playbooks/pod_enrichments.py]: done

File: src/robusta/core/playbooks/container_playbook_utils.py (hunk 5 above)
Comment (ganeshrvel) [reply src/robusta/core/playbooks/container_playbook_utils.py]: done

File: src/robusta/core/playbooks/prometheus_enrichment_utils.py (hunk 6 above)
Comment (ganeshrvel) [reply src/robusta/core/playbooks/prometheus_enrichment_utils.py:217]: as discussed we need this plot_list for adding a line to the chart

**Review by Avi-Robusta — APPROVED** [review]
looks good, can you put a screenshot in the PR before you merge so we have it for the release notes

**Pull request merged**